state/protocol/badger/mutator.go

```diff
@@ -629,6 +637,34 @@ func (m *FollowerState) Finalize(ctx context.Context, blockID flow.Identifier) e
 		m.metrics.EpochEmergencyFallbackTriggered()
 	}
 
+	// FOUR AND A HALF: Index sealed service events by height
+	for _, seal := range block.Payload.Seals {
```

Comment:
Not really - we iterate there only over `parent.Seals`, and we want to index all the blocks.

Reply:
Hi Jordan! Thanks for very detailed answer, it clarified a lot, but I still have some questions.

> that service event comes into effect in the protocol state

Is "comes into effect" actually defined? It seems like it would depend on the even type. Or not?

> ake Leo's suggestion of moving this logic into Step 4

Do you both mean step 3? I don't see how this fit into step 4.

Reply:
Ok, I have applied suggestions